opty/direct_collocation.py

```diff
@@ -244,9 +249,81 @@ def solve(self, free, lagrange=[], zl=[], zu=[]):
                 gives the status of the algorithm as a message
 
         """
-
+        if respect_bounds:
+            self.bounds_conflict_initial_guess(free)
         return super().solve(free, lagrange=lagrange, zl=zl, zu=zu)
 
+    def bounds_conflict_initial_guess(self, free):
```

Comment:
```suggestion
    def check_bounds_conflict(self, free):
```

I suggest this name. It doesn't have to be an initial guess. The action is to check it and raise if there is a conflict.

Reply:
> I suggest this name. It doesn't have to be an initial guess. The action is to check it and raise if there is a conflict.

Definitely better! already changed.